src/hints/execution.rs

```diff
@@ -794,6 +888,53 @@ pub fn cache_contract_storage(
     Ok(())
 }
 
+#[allow(unused)]
+pub const ENTER_SCOPE_NEW_NODE: &str = indoc! {r#"
+	ids.child_bit = 0 if case == 'left' else 1
+	new_node = left_child if case == 'left' else right_child
+	vm_enter_scope(dict(node=new_node, **common_args))"#
+};
+
+pub fn enter_scope_new_node(
+    vm: &mut VirtualMachine,
+    exec_scopes: &mut ExecutionScopes,
+    ids_data: &HashMap<String, HintReference>,
+    ap_tracking: &ApTracking,
+    _constants: &HashMap<String, Felt252>,
+) -> Result<(), HintError> {
+    let left_child: UpdateTree<StorageLeaf> = exec_scopes.get(vars::scopes::LEFT_CHILD)?;
+    let right_child: UpdateTree<StorageLeaf> = exec_scopes.get(vars::scopes::RIGHT_CHILD)?;
+    let case: DecodeNodeCase = exec_scopes.get(vars::scopes::CASE)?;
+
+    let (child_bit, new_node) = match case {
+        DecodeNodeCase::Left => (Felt252::ZERO, left_child),
+        _ => (Felt252::ONE, right_child),
+    };
+
+    insert_value_from_var_name(vars::ids::CHILD_BIT, child_bit, vm, ids_data, ap_tracking)?;
+
+    // vm_enter_scope(dict(node=new_node, **common_args))"#
+    // In this implementation we assume that `common_args` is unpacked, having a
+    // `HashMap<String, Box<dyn Any>>` as scope variable is unpractical.
+    // `common_args` contains the 3 variables below and is never modified.
```

Comment:
Where common_args are declared?

Reply:
Here: https://github.com/starkware-libs/cairo-lang/blob/706ae5fc50663db380b793359e6c6781ffd9ca8c/src/starkware/cairo/common/patricia_with_sponge.cairo#L501

They use a recursive dictionary (`common_args` contains itself) to make it easier to enter new scopes, this is something that I'd avoid in Rust because it will add nothing but complexity.